jx/ui/FillText.jxml

```diff
@@ -2,11 +2,19 @@
 	<script>
 		<![CDATA[
 			function onAttr(attr, delta) {
-				if ( delta.text || attr.text
-					&& (delta.height || attr.height) ) {
-					delta.fontSize = delta.height || attr.height || 12;
+				if (attr.text && attr.height && attr.width) {
+					var ratio = estimate_ratio(attr.text);
+
+					if ( ratio <= (attr.width / attr.height) )
+						delta.fontSize = attr.height;
+					else
+						delta.fontSize = attr.width / ratio;
 				}
 			}
+
+			function estimate_ratio(str) {
```

Comment:
could we have a TODO somewhere if this is something we want to improve on? cc @yangbin